jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/message/MessageInputStream.java

```diff
@@ -258,8 +243,6 @@ public int read(byte[] b, int off, int len) throws IOException
                 SuspendToken resume = null;
                 synchronized (this)
                 {
-                    // Release buffer back to pool.
-                    bufferPool.release(activeBuffer);
```

Comment:
So that actual fix would really just be:
```java
if (activeBuffer != null && activeBuffer != EOF)
    bufferPool.release(activeBuffer);
```
This PR is actually avoiding using a buffer pool at all.